### PR TITLE
Fix deeply nested module imports

### PR DESCRIPTION
**Pull request by Electroid**
In modules mode, when importing a file in a directory, it would get uploaded incorrectly.

`index.mjs`:
```js
import file from "./assets/file.txt"
```

`FormData`:
Before: `file.txt => "..."`
Now: `assets/file.txt => "..."`

### CODE REVIEW

File: src/upload/form/mod.rs [hunk 1]
```diff
@@ -117,8 +117,14 @@ pub fn build(
                         let entry = entry?;
                         let path = entry.path();
                         if path.is_file() {
-                            log::info!("Adding module {}", path.display());
-                            modules.push(Module::new(path.to_owned())?);
+                            let name = path
+                                .strip_prefix(config.upload_dir.clone())?
+                                .to_owned()
+                                .as_path()
+                                .display()
```
Comment (xortive) [inline src/upload/form/mod.rs:124]: This should use the `to_str` method of path and return an error if the conversion fails (due to non-UTF-8 in the path name)

File: src/upload/form/mod.rs [hunk 2]
```diff
@@ -117,8 +117,14 @@ pub fn build(
...
+                                .to_string();
+                            log::info!("Adding module {}", name);
```
Comment (xortive) [inline src/upload/form/mod.rs:126]: would be nice to include the path here as well

File: src/upload/form/mod.rs (hunk 1 above)
Comment (xortive) [reply src/upload/form/mod.rs:124]: presumably for a path like like

`/home/matt/worker/dist/nested/nested/index.js`

This returns `nested/nested/index.js`?

**Review by xortive — CHANGES_REQUESTED** [review]
Nice catch, just handle the UTF-8 thing and I'll merge

**Comment by Electroid** [comment]
Realized the previous approach would not work on Windows machines, so had to add a dependency to help us out.

**Comment by xortive** [comment]
@Electroid could you rebase this on the `malonso/modules-rc` branch? I split that out from the old `modules-do` branch, and it looks like this PR is trying to add the durable objects support back in along with your fixes.

**Review by xortive — APPROVED** [review]
Thanks @Electroid!

**Pull request merged**